### ISSUE TITLE
Code - Implement data augmentation techniques 

### ISSUE THREAD

**Issue by tristaaan** [diegocasmo/ml-project#25]
~~It obviously takes a while to fully train this, as fun as it is babysitting my computer for hours on end I would like to plan to do runs over the course of a night.~~ This gives us a few chances to generate some new results. Right now, I think there are a few ideas to try. Here are a some:

- train the current iteration for another 50 epochs
- data augmentation with gaussian noise
- HSL values 

These are certainly up for debate, but I’d argue we only have a few nights to try this. Any other ideas?

**Comment by diegocasmo** [comment]
- [x] batches of size 16
- [x] crop down to 207x207
- [x] downsample 3x to 69x69
- [ ] rotate images by random multiple of 90

**Comment by tristaaan** [comment]
With the dataset preprocessed to 106x106, 50 epochs should take around 2 hours. So I've redacted a bit of the urgency in the initial issue comment.

**Comment by tristaaan** [comment]
> rotate images by random multiple of 90

https://keras.io/preprocessing/image/#random_transform

**Comment by tristaaan** [comment]
Runs I have planned right now: 

- [x] 106x106 with batches of 32
- [x] 106x106 with batches of 16, ~~very fast training, early stopped at 39 epochs, not a good Kaggle score: **0.29801** which is pretty bad.~~
- [x] 69x69 batches of 32: ~~**0.33997**, this is almost the worst possible score.~~ 
- [x] 69x69 batches of 16: ~~I'm skipping this one right now, the loss graphs suggest we're severely over-fitting, or there is some other issue :|~~

The 69x69 are cropped 207 then re-sampled to 69x69.

**Comment by tristaaan** [comment]
Scores are good now, taking suggestions for other runs

- [x] Data augmentation: random rotations
- [x] Process images with HSV values (no results yet, but loss graph looks mediocre)

**Comment by tristaaan** [comment]
The code is in a pretty good state, we should not be augmenting the validation set. I'm trying two more runs for effects of augmentation on batch_size and then I'm done.

**Comment by tristaaan** [comment]
Closed with #28
